src/services/signUpSignInFunction.js

```diff
@@ -0,0 +1,49 @@
+import openNotification from "../components/ReusableComponents/Notification";
+import Cookies from "js-cookie";
+import { postRequest } from "./fetchUtils";
+
+const signUpSignInFunction = ({ response, history, loginData = {} }) => {
+  const isError = response && "err" in response;
+  const isEmailVerifiedFalse =
+    response && response.token && response.emailVerified === false;
+  const isEmailVerifiedTrue =
+    response && response.token && response.emailVerified === true;
+  const isSignUpSuccess =
+    response &&
+    response.success &&
+    "email" in loginData &&
+    "password" in loginData;
+
+  switch (true) {
+    case isError:
+      return openNotification({
+        type: "error",
+        message: "Ошибка",
+        description: response.err
+      });
+    case isEmailVerifiedFalse:
+      Cookies.set("Access-Token", response.token);
+      return history.push("/email-not-verified");
+    case isEmailVerifiedTrue:
+      Cookies.set("Access-Token", response.token);
+      return history.push("/");
```

Comment:
case validToken:
Cookies.set("Access-Token", response.token);
rerurn history.push(response.emailVerified ? "/" : "/email-not-verified")